inference-engine/src/transformations/src/transformations/preprocessing/scale_inputs.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+#include <array>
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <unordered_map>
+#include <unordered_set>
+
+#include <ngraph/variant.hpp>
+#include "ngraph/ops.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include <ngraph/opsets/opset7.hpp>
+#include "ngraph_ops/framework_node.hpp"
+#include "pugixml.hpp"
+#include "transformations/preprocessing/scale_inputs.hpp"
+#include "ngraph/pattern/op/wrap_type.hpp"
+#include <ngraph/pattern/op/or.hpp>
+
+using namespace ngraph;
+using namespace ngraph::pass;
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ScaleInputs, "ScaleInputs", 0);
+
+using ConstantCreator = std::function<std::shared_ptr<opset7::Constant>(std::shared_ptr<Node>)>;
+
+static bool matcher_callback(pattern::Matcher& m, ConstantCreator creator) {
+    auto node = m.get_match_root();
+    auto consumers = node->output(0).get_target_inputs();
+    auto mul_const = creator(node);
+    mul_const->set_friendly_name(node->get_friendly_name() + "/scale/Fused_Mul_Factor");
+    auto new_op = std::make_shared<ngraph::opset7::Multiply>(node, mul_const);
+    new_op->set_friendly_name(node->get_friendly_name() + "/scale/Fused_Mul");
+    for (auto consumer : consumers) {
+        consumer.replace_source_output(new_op);
+    }
+    return true;
+}
+
+void ScaleInputs::register_scale_matcher(ngraph::matcher_pass_callback callback) {
+    auto param = pattern::wrap_type<op::Parameter>();
+    auto m = std::make_shared<ngraph::pattern::Matcher>(param, "ScaleMatcher");
+    register_matcher(m, callback);
+}
+
+ScaleInputs::ScaleInputs(float scale_factor): MatcherPass(), m_scale_factor(scale_factor) {
+    register_scale_matcher(std::bind(matcher_callback, std::placeholders::_1, [&](std::shared_ptr<Node>) {
+        return opset7::Constant::create(ngraph::element::f32, ngraph::Shape{1}, {1.f / m_scale_factor});
+    }));
+}
+
+/// Calculate features dimension index based on input info
+/// If initial_idx is not -1, use it if dimension equals to values_size
+/// E.g. node_shape = {1,3,224,224}, scale_size=3 ==> Result will be "1" - dimension #1 is a 'features index'
+static int guess_features_dim_idx(const std::shared_ptr<Node>& matched, size_t values_size, int initial_idx) {
```

Comment:
TODO: this should be moved somewhere to be reusable by future 'reduce_mean_inputs' transformation